src/spi.rs

```diff
@@ -464,14 +566,25 @@ impl<'d> SpiMasterDriver<'d> {
     ) -> Result<R, E>
     where
         E: From<EspError>,
+        T: Borrow<SpiMasterDriver<'d>> + 'd,
```

Comment:
Can this be moved to the `impl` block? Any reason to do it on every method?

Reply:
yeah this is is a bit awkward. i could't get the drop fn otherwise working :D

Reply:
but i can make it work i guess by getting the handle directly and dont rely on a rm_from_bus Fn inside Drop

Reply:
Yeah, at the very least you can inline the implementation of `rm_from_bus` to where it's used. (I'm still of the opinion that it should be removed though)

Reply:
done